### ISSUE TITLE
Weierstrass panic: Points are the same. Use sw_double instead

### ISSUE THREAD

**Issue by yaziciahmet** [succinctlabs/sp1#1680]
Hey guys!

I am receiving the following panic when I run my zk program in my MacOS machine locally, only execution:

```
2024-10-18T18:25:43.958973Z ERROR execute: panic: thread '<unnamed>' panicked at 'Error: Points are the same. Use sw_double instead.': ~/.cargo/git/checkouts/sp1-20c98843a1ffc860/59c4234/crates/curves/src/weierstrass/mod.rs:176
```

I delved deeper into it and figured that it happens where I call `TransactionSigned::into_ecrecovered` in `reth_primitives::transaction` crate. I am using commit [a206eb3690e5a51d3c797fed2a6ed722e36863eb](https://github.com/paradigmxyz/reth/tree/a206eb3690e5a51d3c797fed2a6ed722e36863eb).

I use this method multiple times, but it happens for the following specific transaction:

```
TransactionSigned { hash: 0x94dc0ce2952f04b9224e43322717784e177320fbfc3bdcb295472437bcb7ddce, signature: Signature { r: 65380941964943506365915457270296004249787913312232689974263100638689464413246, s: 42584810713712003418817999545753212719523848378983378602657690381670578672061, odd_y_parity: true }, transaction: Legacy(TxLegacy { chain_id: Some(5115), nonce: 808, gas_price: 2146698501, gas_limit: 2950000, to: Call(0x62ef91b746f9fd31adac6c211ad8b8213c1c9ba4), value: 0, input: 0xb189fd4c11074d1e3b6566dedd5883b35c4622daeb03b324e31d5ee40a55e62bad558510 }) }
```

I am using sp1-sdk and sp1-zkvm v3.0.0-rc3. 

I apply the following patches to my program:

```
sha2-v0-10-8 = { git = "https://github.com/sp1-patches/RustCrypto-hashes", package = "sha2", branch = "patch-sha2-v0.10.8" }
sha3-v0-10-8 = { git = "https://github.com/sp1-patches/RustCrypto-hashes", package = "sha3", branch = "patch-sha3-v0.10.8" }
crypto-bigint = { git = "https://github.com/sp1-patches/RustCrypto-bigint", branch = "patch-v0.5.5" }
tiny-keccak = { git = "https://github.com/sp1-patches/tiny-keccak", branch = "patch-v2.0.2" }
curve25519-dalek = { git = "https://github.com/sp1-patches/curve25519-dalek", branch = "patch-curve25519-v4.1.3" }
ecdsa-core = { git = "https://github.com/sp1-patches/signatures", package = "ecdsa", branch = "patch-ecdsa-v0.16.9" }
secp256k1 = { git = "https://github.com/sp1-patches/rust-secp256k1", branch = "patch-secp256k1-v0.29.1" }
substrate-bn = { git = "https://github.com/sp1-patches/bn", branch = "patch-v0.6.0" }
bls12_381 = { git = "https://github.com/sp1-patches/bls12_381", branch = "patch-v0.8.0" }
```

**Comment by ratankaliani** [comment]
Hey @yaziciahmet! We recently resolved this issue by updating our patch for `ecdsa-core` to use the latest version of SP1. Can you retry by running `cargo update -p ecdsa@0.16.9` to update to the latest version of the patch? 

This is the relevant issue where we included the changes: https://github.com/succinctlabs/sp1/pull/1542.

After you do so, this issue should be fixed! I'd also recommend updating to SP1 `3.0.0`.

**Comment by yaziciahmet** [comment]
Thanks a lot @ratankaliani! It fixed the issue.
